docs/CImportNameTranslation.md

```diff
@@ -0,0 +1,191 @@
+This document gives a high-level description of how C and Objective-C declarations are translated to Swift, with particular focus on how names are adjusted. It is not attempting to be a *complete* description of everything the compiler does except with regards to how *names* are transformed; even there, some special cases that only apply to Apple's SDKs have been omitted.
```

Comment:
This should probably have a title.

Reply:
I'm always conflicted about that because the filename is sort of a title, but sure.